src/checkers/inference/solver/backend/maxsat/MaxSatSolver.java

```diff
@@ -115,14 +109,59 @@ public MaxSatSolver(SolverEnvironment solverEnvironment, Collection<Slot> slots,
                 // PrintUtils.printResult(result);
             } else {
                 System.out.println("Not solvable!");
+                result = null;
             }
 
-        } catch (Throwable e) {
-            e.printStackTrace();
+        } catch (ContradictionException e) {
+            InferenceMain.getInstance().logger.warning("Contradiction exceptin: ");
+            // This case indicates that constraints are not solvable, too. This is normal so continue
+            // execution and let solver strategy to explain why there is no solution
+            result = null;
```

Comment:
Set `result` to `null` so that `InferenceResult#hasSolution()` returns false.

Reply:
Could this method initialize `result` with `null` and only assign solution to it when having solutions? In this case, for any other abnormal branches `result` still be `null` and we don't need to re-assign `result` to `null` in every branch for abnormal cases.

Reply:
Sounds good idea. Will do that.